src/sap.ui.core/src/sap/ui/core/date/Islamic.js

```diff
@@ -245,9 +245,9 @@ sap.ui.define([
 
 	function parseDate(sDate) {
 		return {
-			year: parseInt(sDate.substr(0, 4)),
-			month: parseInt(sDate.substr(4, 2)),
-			day: parseInt(sDate.substr(6, 2))
+			year: parseInt(sDate.substring(0, 4)),
+			month: parseInt(sDate.substring(4, 2)),
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr says "The two parameters of substr() are start and length, while for substring(), they are start and end.". How is a simple method name replacement supposed to work in case start is not zero?